### PR TITLE
Add a small lambda example in methods.md

### PR DESCRIPTION
**Pull request by codec-abc**

### CODE REVIEW

File: expressions/methods.md [hunk 1]
```diff
@@ -228,6 +228,32 @@ primitive Foo
     f.>add_option(o1).>add_option(o2).make_object() // Works. The expression returns an Object
 ```
 
+## Anonymous methods
+
+Pony has Anonymous methods (or Lambdas). They look like this :
+
+```pony
+/* Print
```
Comment (mfelsche) [inline expressions/methods.md]: I'd remove those comments.

File: expressions/methods.md [hunk 2]
```diff
@@ -228,6 +228,32 @@ primitive Foo
...
+    let filteringLambda = {(n: U32): Bool => n % 2 == 1}
+    let onlyOdd = listOfNumbers.filter(filteringLambda).values()
+    while onlyOdd.has_next() do
+      try
+        env.out.print(onlyOdd.next()?.string())
+      end
+    end
+```
+
+They are presented more in-depth in the Object Literals section.
```
Comment (mfelsche) [inline expressions/methods.md]: Pls make `Object Literals section` a link to the actual section: `(Object Literals section)[../expressions/object-literals.md]`

File: expressions/methods.md [hunk 3]
```diff
@@ -228,6 +228,32 @@ primitive Foo
     f.>add_option(o1).>add_option(o2).make_object() // Works. The expression returns an Object
 ```
 
+## Anonymous methods
+
+Pony has Anonymous methods (or Lambdas). They look like this :
```
Comment (mfelsche) [inline expressions/methods.md]: Remove whitespace between `this` and `:`

Comment (mfelsche) [inline expressions/methods.md]: typo: `anonymous` (lowercase)

File: expressions/methods.md (hunk 2 above)
Comment (codec-abc) [reply expressions/methods.md]: Shouldn't be [Object Literals section](../expressions/object-literals.md) instead (swapping parenthesis and square brackets)

**Pull request merged**